src/crimson/os/seastore/seastore_types.cc

```diff
@@ -512,19 +512,22 @@ ceph::bufferlist encode_record(
   record_t&& record,
   extent_len_t block_size,
   const journal_seq_t& committed_to,
-  segment_nonce_t current_segment_nonce)
+  segment_nonce_t current_segment_nonce,
+  bool checksum_offloaded_to_device)
 {
   record_group_t record_group(std::move(record), block_size);
   return encode_records(
       record_group,
       committed_to,
-      current_segment_nonce);
+      current_segment_nonce,
+      checksum_offloaded_to_device);
 }
 
 ceph::bufferlist encode_records(
   record_group_t& record_group,
   const journal_seq_t& committed_to,
-  segment_nonce_t current_segment_nonce)
+  segment_nonce_t current_segment_nonce,
+  bool checksum_offloaded_to_device)
```

Comment:
Same here, I don't think device crc offloading is related to record crc, records can still become partially written under failure, and seastore needs to detect this case.